pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -369,31 +374,37 @@ func (r *ReconcileBareMetalHost) credentialsErrorResult(err error, request recon
 	// at some point in the future.
 	case *ResolveBMCSecretRefError:
 		credentialsMissing.Inc()
-		changed, saveErr := r.setErrorCondition(request, host, metal3v1alpha1.RegistrationError, err.Error())
+		changed, requeue, saveErr := r.setErrorCondition(request, host, metal3v1alpha1.RegistrationError, err.Error())
```

Comment:
What if instead of returning 3 values from this function so they have to be checked everywhere, we return a reconcile.Result{} and an error? If either is not nil, return both, otherwise drop through.

Reply:
We shouldn't have to look inside it, only check if it's nil or not, right?

Reply:
If it returned `*reconcile.Result` I guess so, yeah. Elsewhere we only pass/return `Result`s by value.

Reply:
Ah, yeah, good point.